lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -366,16 +367,20 @@ export class UserAgentApplication {
       // TODO: Should we throw noCallback error here?
     }
 
-    // TODO: Replace with new validation pattern
-    if (scopes) {
-      const isValidScope = this.validateInputScope(scopes);
-      if (isValidScope && !Utils.isEmpty(isValidScope)) {
-          if (this._tokenReceivedCallback) {
-              this._tokenReceivedCallback(ErrorDescription.inputScopesError, null, ErrorCodes.inputScopesError, Constants.idToken, this.getUserState(this._cacheStorage.getItem(Constants.stateLogin, this.storeAuthStateInCookie)));
+    try {
+      scopes = this.validateAndFilterInputScope(scopes);
+    } catch (err) {
+      if (err instanceof ClientConfigurationError) {
+        if (this._tokenReceivedCallback) {
+          this._tokenReceivedCallback(ErrorDescription.inputScopesError, null, ErrorCodes.inputScopesError, Constants.idToken, this.getUserState(this._cacheStorage.getItem(Constants.stateLogin, this.storeAuthStateInCookie)));
           return;
+        } else {
+          // We should be checking for callback earlier
```

Comment:
should we throw a more specific type of error here? Configuration error?

Reply:
There will be a function that checks for redirect callbacks in a future PR